### PR TITLE
Use latest and concept DOI in docs instead of the specific one

### PR DESCRIPTION
**Pull request by francisco-dlp**
This is part of the ongoing effort to streamline the release process. See discussion in #2490

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -53,7 +56,4 @@ Cite
 
 If you use HyperSpy in a scientific publication, please cite the following DOI:
```
Comment (ericpre) [inline README.rst:57]: Either this sentence is removed or the DOI is also included after but it can't be in between! ;)

**Review by tjof2 — APPROVED** [review]

File: doc/citing.rst [hunk 2]
```diff
@@ -3,8 +3,12 @@
 ================
 
 If HyperSpy has been significant to a project that leads to an academic
-publication, please acknowledge that fact by citing the software. Click on
-the DOI badge below for formatted citation formats.
+publication, please acknowledge that fact by citing it. The DOI in the
+badge below is the Concept DOI of HyperSpy. It can be used to cite the project
```
Comment (ericpre) [inline doc/citing.rst:7]: `Concept DOI (for all versions)` because it isn't obvious.
Add link to https://help.zenodo.org ?

**Review by ericpre — COMMENTED** [review]
A few tweaks

File: README.rst (hunk 1 above)
Comment (tjof2) [reply README.rst:57]: Ah you reviewed it at the same time I hit merge :-/

Comment (francisco-dlp) [reply README.rst:57]: Thanks done

File: doc/citing.rst (hunk 2 above)
Comment (francisco-dlp) [reply doc/citing.rst:7]: Done

**Comment by francisco-dlp** [comment]
Shall I merge the commits (f3dc3ea433319e1 and 5c73badc84b47 ) directly into Rnp?

**Comment by tjof2** [comment]
Sounds reasonable

**Comment by francisco-dlp** [comment]
Done.

**Pull request merged**